tests/cowswap.spec.ts

```diff
@@ -40,42 +40,235 @@ const quoteRequest = {
 };
 
 describe("CowSwap Plugin", () => {
-  // This posts an order to COW Orderbook.
-  it.skip("orderRequestFlow", async () => {
+  // This posts an order to COW Orderbook and validates the response format
+  it("orderRequestFlow returns valid swap data", async () => {
     console.log("Requesting Quote...");
-    const signRequest = await orderRequestFlow({
+    const response = await orderRequestFlow({
       chainId,
       quoteRequest: { ...quoteRequest, from: DEPLOYED_SAFE },
+      buyTokenData: { address: SEPOLIA_COW, decimals: 18, symbol: "COW" },
+      sellTokenData: { address: SEPOLIA_DAI, decimals: 18, symbol: "DAI" },
     });
-    console.log(signRequest);
+
+    // Verify the data property has the correct structure
+    expect(response.data).toBeDefined();
+    expect(response.data).toHaveProperty("network");
+    expect(response.data.network).toHaveProperty("name", chainId.toString());
+    expect(response.data).toHaveProperty("type", "swap");
+
+    // Verify token information
+    expect(response.data).toHaveProperty("tokenIn");
+    expect(response.data.tokenIn).toHaveProperty("name", "DAI");
+    expect(response.data.tokenIn).toHaveProperty("amount");
+
+    expect(response.data).toHaveProperty("tokenOut");
+    expect(response.data.tokenOut).toHaveProperty("name", "COW");
+    expect(response.data.tokenOut).toHaveProperty("amount");
+
+    // Validate amount formats and values
+    expect(typeof response.data.tokenIn.amount).toBe("string");
+    expect(typeof response.data.tokenOut.amount).toBe("string");
+
+    // Check if the amount exists but possibly in a different format
+    expect(response.data.tokenIn.amount).toBeTruthy();
+    expect(response.data.tokenOut.amount).toBeTruthy();
+
+    // We now know the API returns a formatted amount like "2" instead of "2000000000000000000"
+    // So we can't directly compare with the input
+    const inAmount = Number.parseFloat(response.data.tokenIn.amount);
+    expect(inAmount).toBe(2); // 2 ETH
+
+    // Check that tokenOut amount is a valid positive number
+    const outAmount = Number.parseFloat(response.data.tokenOut.amount);
+    expect(outAmount).toBeGreaterThan(0);
+
+    // Validate fee information if present
+    if ("fee" in response.data) {
+      const fee = response.data.fee as { amount: string };
+      expect(fee).toHaveProperty("amount");
+      expect(typeof fee.amount).toBe("string");
+      expect(/^\d+$/.test(fee.amount)).toBe(true);
+    }
+
+    // Verify the final response has a transaction property
+    expect(response).toHaveProperty("transaction");
+
+    // Validate transaction object structure
+    const tx = response.transaction as unknown as {
+      chainId: number;
+      method: string;
+      params: Array<{
+        to: string;
+        data: string;
+        from: string;
+        value: string;
+      }>;
+    };
```

Comment:
This does not need to be cast to a type.
```suggestion
    const tx = response.transaction;
```